tests/nonce/test_nonce_utils.cairo

```diff
@@ -0,0 +1,89 @@
+use starknet::{
+    ContractAddress, get_caller_address, Felt252TryIntoContractAddress, contract_address_const
+};
+use snforge_std::{declare, start_prank, stop_prank, ContractClassTrait};
+
+use satoru::data::data_store::{IDataStoreSafeDispatcher, IDataStoreSafeDispatcherTrait};
+use satoru::role::role_store::{IRoleStoreSafeDispatcher, IRoleStoreSafeDispatcherTrait};
+use satoru::role::role;
+use satoru::nonce::nonce_utils::{get_current_nonce, increment_nonce, get_next_key};
+
+#[test]
+fn test_nonce_utils() {
+    // *********************************************************************************************
+    // *                              SETUP                                                        *
+    // *********************************************************************************************
+    let data_store = setup();
+
+    // *********************************************************************************************
+    // *                              TEST LOGIC                                                   *
+    // *********************************************************************************************
+
+    let nonce = get_current_nonce(data_store).unwrap();
+    assert(nonce == 0, 'Invalid nonce');
+
+    let nonce = increment_nonce(data_store).unwrap();
+    assert(nonce == 1, 'Invalid new nonce');
+
+    let key = get_next_key(data_store).unwrap();
+    assert(key == 0x3f84fbc06ce0aca2f042f92dbe31a1426167c15392bba1e905ec3c3f0c177f7, 'Invalid key');
+
+    let nonce = get_current_nonce(data_store).unwrap();
+    assert(nonce == 2, 'Invalid final nonce');
+
+    // *********************************************************************************************
+    // *                              TEARDOWN                                                     *
+    // *********************************************************************************************
+    teardown(data_store.contract_address);
+}
+
+/// Utility function to deploy a data store contract and return its address.
+///
+/// # Arguments
+///
+/// * `role_store_address` - The address of the role store contract.
+///
+/// # Returns
+///
+/// * `ContractAddress` - The address of the deployed data store contract.
+fn deploy_data_store(role_store_address: ContractAddress) -> ContractAddress {
```

Comment:
there should be a way to reuse this code from `test_data_store.cairo` using snforge, it was possible with vanilla scarb

Reply:
yes, probably but havent find it yet haha. If you discover how please update us!